lib/active_merchant/billing/gateways/datatrans.rb

```diff
@@ -86,10 +108,23 @@ def scrub(transcript)
       private
 
       def add_payment_method(post, payment_method)
-        card = build_card(payment_method)
+        card = {}
```

Comment:
💬 Comment:

I see this as an opportunity to refactor `add_payment_method` to be more explicit/declarative on the payment method construction, something like:

```ruby
def add_payment_method(post, payment_method)
  case payment_method
  when String
    token, exp_month, exp_year = payment_method.split('|')[2..4]
    card = {
      type: 'ALIAS',
      alias: token,
      expiryMonth: exp_month,
      expiryYear: exp_year
    }
  when NetworkTokenizationCreditCard
    card = {
      type: DEVICE_SOURCE[payment_method.source] ? 'DEVICE_TOKEN' : 'NETWORK_TOKEN',
      tokenType: DEVICE_SOURCE[payment_method.source] || CREDIT_CARD_SOURCE[card_brand(payment_method)],
      token: payment_method.number,
      cryptogram: payment_method.payment_cryptogram,
      expiryMonth: format(payment_method.month, :two_digits),
      expiryYear: format(payment_method.year, :two_digits)
    }
  when CreditCard
    card = {
      type: 'PLAIN',
      number: payment_method.number,
      cvv: payment_method.verification_value.to_s,
      expiryMonth: format(payment_method.month, :two_digits),
      expiryYear: format(payment_method.year, :two_digits)
    }
  end

  post[:card] = card
end
```